### PR TITLE
day_of_year

### LINKED ISSUE #9606: API: timeseries accessors naming convention
we have a slightly odd naming convention for several index (and .dt) accessors

`dayofweek`
`dayofyear`

(and adding `daysinmonth` in #9605)

so these should prob be `day_of_week` 

we should be able to maintain compat on these and just add the new ones.

### PR DESCRIPTION
**Pull request by oleksii-polovyi**
- [ ] closes #9606
- [ ] tests added / passed
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by mroeschke** [comment]
`dayofyear` cannot be directly replaced with `day_of_year ` all at once. A new `day_of_year` method needs to be created first and then `dayofyear` needs a deprecation warning (all of which can be done in this PR). If you could do the same with `dayofweek` that'd be great.

**Comment by ApolllonDev** [comment]
@mroeschke thank you for your feedback.
TODO:
- [ ] deprecation warnings
- [ ] dayofweek

**Comment by Violet-XiaoWeiHuang** [comment]
Hello @mroeschke , 
I'm a new comer to Pandas, I notice there're other timeseries accessors like 'weekofyear'. May I work on it? 
Thank you.

**Comment by mroeschke** [comment]
Thanks for the interest @Violet-XiaoWeiHuang. Since the fix is the same as these other accessors, I'll give @ApolllonDev the choice whether to work on the other accessors as well or pass it on to you.

**Comment by ApolllonDev** [comment]
That would be great!

File: doc/source/whatsnew/v0.16.2.rst [hunk 1]
```diff
@@ -137,7 +137,7 @@ Bug Fixes
 - Bug in ``mean()`` where integer dtypes can overflow (:issue:`10172`)
 - Bug where ``Panel.from_dict`` does not set dtype when specified (:issue:`10058`)
 - Bug in ``Index.union`` raises ``AttributeError`` when passing array-likes. (:issue:`10149`)
-- Bug in ``Timestamp``'s' ``microsecond``, ``quarter``, ``dayofyear``, ``week`` and ``daysinmonth`` properties return ``np.int`` type, not built-in ``int``. (:issue:`10050`)
+- Bug in ``Timestamp``'s' ``microsecond``, ``quarter``, ``day_of_year``, ``week`` and ``daysinmonth`` properties return ``np.int`` type, not built-in ``int``. (:issue:`10050`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.16.2.rst]: don't change prior whatsnews

File: pandas/core/arrays/datetimes.py [hunk 2]
```diff
@@ -1368,8 +1368,8 @@ def date(self):
         'weekday_name',
         "The name of day in a week (ex: Friday)\n\n.. deprecated:: 0.23.0")
 
-    dayofyear = _field_accessor('dayofyear', 'doy',
-                                "The ordinal day of the year.")
+    day_of_year = _field_accessor('day_of_year', 'doy',
```
Comment (jreback) [inline pandas/core/arrays/datetimes.py]: we need to preserve the original with a deprecation warning

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/arrays/datetimes.py (hunk 2 above)
Comment (ApolllonDev) [reply pandas/core/arrays/datetimes.py]: I will add deprecation warnings on the weekend. Thank you for your feedback.

**Comment by jreback** [comment]
pls update per comments

File: doc/source/whatsnew/v0.16.2.rst (hunk 1 above)
Comment (ApolllonDev) [reply doc/source/whatsnew/v0.16.2.rst]: Fixed.

**Pull request closed without merging**